### PR TITLE
Delete user

### PR DESCRIPTION
**Pull request by rhnonose**
Soft delete user and not show in the index.

### CODE REVIEW

File: lib/user_role.ex [hunk 1]
```diff
@@ -0,0 +1,27 @@
+defmodule CoursePlanner.Types.UserRole do
+  @moduledoc """
+    This module introduces a custom type for Etco for checking user role in the user model
+  """
+  @behaviour Ecto.Type
+  def type, do: :user_role
+
+  @valid_values ["Student", "Teacher", "Organizer"]
```
Comment (ghatighorias) [inline lib/user_role.ex]: we need "volenteer" too :)

Comment (rhnonose) [inline lib/user_role.ex]: right :)

**Comment by ghatighorias** [comment]
Quick question : is it possible to remove all the organizer users ?
cause if so then it will be problematic ( an app which won't have any admin )

**Comment by rhnonose** [comment]
Maybe we can have an admin user as backup, but I don't think it's likely when we setup the permissions.

File: priv/static/js/app.js [hunk 2]
```diff
@@ -1549,8 +1549,8 @@ channel.join().receive("ok", function (resp) {
 exports.default = socket;
 });
 
-;require.alias("phoenix_html/priv/static/phoenix_html.js", "phoenix_html");
-require.alias("phoenix/priv/static/phoenix.js", "phoenix");require.register("___globals___", function(exports, require, module) {
+;require.alias("phoenix/priv/static/phoenix.js", "phoenix");
```
Comment (ggpasqualino) [inline priv/static/js/app.js]: can we ignore this file? it keeps coming and going when it is regenerated

Comment (rhnonose) [inline priv/static/js/app.js]: Yeah, I still don't know what's the problem.

File: priv/repo/migrations/20170424091310_soft_delete_user.exs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+defmodule CoursePlanner.Repo.Migrations.SoftDeleteUser do
+  use Ecto.Migration
+
+  def change do
+    alter table(:users) do
+      add :deleted, :boolean
+      add :deleted_at, :utc_datetime
```
Comment (rhnonose) [inline priv/repo/migrations/20170424091310_soft_delete_user.exs]: Good idea.

File: priv/repo/migrations/20170424091310_soft_delete_user.exs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+defmodule CoursePlanner.Repo.Migrations.SoftDeleteUser do
+  use Ecto.Migration
+
+  def change do
+    alter table(:users) do
+      add :deleted_at, :utc_datetime
```
Comment (ggpasqualino) [inline priv/repo/migrations/20170424091310_soft_delete_user.exs]: I was talking to @ghatighorias about this type. What do you think of using `naive_datetime`?
The standard timestamps (inserted_at, updated_at) use this type

Comment (rhnonose) [inline priv/repo/migrations/20170424091310_soft_delete_user.exs]: I don't mind. Phoenix is going to support utc_datetime generators, but the standard will be naive_datetime.
Unless we're going to deal with timezones, we can set all to naive_datetime.

Comment (ggpasqualino) [inline priv/repo/migrations/20170424091310_soft_delete_user.exs]: okay, I say that we make it naive then, to keep it consistent with the others

Comment (rhnonose) [inline priv/repo/migrations/20170424091310_soft_delete_user.exs]: One issue though, coherence has utc_datetime types.
The effort to keep it consistent should be higher for naive_datetime.

**Comment by ggpasqualino** [comment]
can you create a new pr without this user roles things in it? 
github doesn't seem to get the diff properly after I merged the other pr about roles

**Pull request closed without merging**